### PR TITLE
Minor optimization when installing a language

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | During the database content translation process (which happens whenever a new language is installed), some operations where needlessly being done before checking if there was effectively data to translate. Delaying this until after we have verified that there will be data to translate might speed up the install process.
| Type?         | refacto
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | N/A
| How to test?  | Install a shop in a language. Add a new language and switch to it. The BO menu should be translated as usual.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17979)

### CODE REVIEW

File: classes/Language.php [hunk 1]
```diff
@@ -1485,15 +1485,7 @@ public static function updateMultilangFromClass($table, $className, $lang)
...
     {
-        $shopDefaultLangId = Configuration::get('PS_LANG_DEFAULT', null, $shop->id_shop_group, $shop->id);
-        $shopDefaultLanguage = new Language($shopDefaultLangId);
-
-        $translator = SymfonyContainer::getInstance()->get('translator');
-        if (!$translator->isLanguageLoaded($shopDefaultLanguage->locale)) {
-            (new TranslatorLanguageLoader(true))->loadLanguage($translator, $shopDefaultLanguage->locale);
-        }
-
-        $shopFieldExists = $primary_key_exists = false;
```
Comment (eternoendless) [inline classes/Language.php:1496]: I removed `$primary_key_exists` because there were no references to it.

File: classes/Language.php [hunk 2]
```diff
@@ -1476,7 +1476,7 @@ public static function updateMultilangFromClass($table, $className, $lang)
      *
      * @param string $tableName
      * @param DataLang $classObject
-     * @param string $lang
```
Comment (eternoendless) [inline classes/Language.php:1479]: This typehint was wrong. Verified using debugger.

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @eternoendless,

It is Ok :heavy_check_mark:

Thanks!

**Comment by Progi1984** [comment]
Thanks @eternoendless

**Pull request merged**